lib/uring/uring.ml

```diff
@@ -237,13 +232,16 @@ let register_gc_root t =
 let unregister_gc_root t =
   update_gc_roots (Ring_set.remove (Generic_ring.T t))
 
-let create ?polling_timeout ~queue_depth () =
+let create ?polling_timeout ?(sketch_size=1024 * 16) ~queue_depth () =
   if queue_depth < 1 then Fmt.invalid_arg "Non-positive queue depth: %d" queue_depth;
+  if sketch_size < 1 then Fmt.invalid_arg "Non-positive sketch_size: %d" sketch_size;
+  if (sketch_size land 0x7) <> 0 then Fmt.invalid_arg "sketch_size must be 8 byte aligned";
```

Comment:
Maybe just round it here instead of complaining, since we already know what the rules are. Does it even matter if the size isn't a round number?

Reply:
I disagree here, I really dislike interfaces where you give a number and it magically changes to somewhere else.
On the sketch_alloc() the allocated value is actually returned so the user at least knows if the value was mangled.

Reply:
I suspect that unit-tests might be the only place where this is useful though. The case I'm thinking about is capnp-rpc, which queues up messages and then submits them all in one go. There could be iovecs with hundreds of entries, and canp-rpc doesn't know anything about uring. Just returning `None` if there are too many isn't useful, because Eio will just wait for more completions and retry, which will fail again. So I'd prefer to have uring make sure the operation works, either by resizing the buffer or by truncating the list. There's no very good reason for a user to care how big the sketch buffer is.

Reply:
Got it, let me summarize what I think we agreed on:

- Remove the `free` operation
- Move the allocation logic up to the ocaml side
- Resize (I prefer this to truncating, less data to keep track of)
- sketch_size goes away if we do resize

Concerning resize, do we want to resize "always up to a limit" or do we want to resize only when the requested allocation is larger than the total allocated (as in, not just larger than avail, larger then the buffer length).